tests/test_git.py

```diff
@@ -1,12 +1,123 @@
+O4pG4qMvNU8DB89yOtPNAAAAA0JDSQEC
+-----END OPENSSH PRIVATE KEY-----"""
+
+PUB_KEY = "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIMkxHs1b6A1kZuiu6nPxehkSO4pG4qMvNU8DB89yOtPN BCI"
+
+GIT_SERVER_CONTAINERFILE = rf"""
+RUN zypper -n in git openssh-server
+RUN ssh-keygen -A
+RUN useradd -U -m -p "*" git
+
+RUN mkdir /home/git/.ssh && touch /home/git/.ssh/authorized_keys
+RUN chmod 700 /home/git/.ssh && chmod 600 /home/git/.ssh/authorized_keys
+RUN chown -R git:git /home/git/.ssh
+RUN echo "{PUB_KEY}" >> /home/git/.ssh/authorized_keys
+
+RUN mkdir -p /srv/git/project.git
+RUN git -C /srv/git/project.git init --bare
+
+RUN echo -e '\
+AuthorizedKeysFile      .ssh/authorized_keys\n\
+ChallengeResponseAuthentication no\n\
+PasswordAuthentication no\n\
+PermitRootLogin no\n\
+PubkeyAuthentication yes\n\
+AllowUsers git\n\
+UsePAM no\n\
+AllowTcpForwarding no\n\
+X11Forwarding no\n\
+Subsystem       sftp    /usr/lib/ssh/sftp-server\n\
+' > /etc/ssh/sshd_config
+
+EXPOSE 22
+
+CMD ["/usr/sbin/sshd", "-De"]
+"""
+
 
 def test_git_version(auto_container):
     assert "git version 2." in auto_container.connection.check_output(
         "git --version"
     )
+
+
+def test_git_clone_https(auto_container_per_test):
+    """Test that we can clone a repository over HTTPS successfully."""
+    auto_container_per_test.connection.run_expect(
+        [0], "git clone https://github.com/github/gemoji.git /gemoji"
+    )
+    auto_container_per_test.connection.file("/gemoji/.git/HEAD").exists
+    auto_container_per_test.connection.file("/gemoji/Gemfile").exists
+    auto_container_per_test.connection.file("/gemoji/db/emoji.json").exists
+    out = auto_container_per_test.connection.check_output(
+        "git -C /gemoji status"
+    )
+    assert "Your branch is up to date" in out
+    assert "nothing to commit, working tree clean" in out
+
+
+GIT_SERVER_CONTAINER = DerivedContainer(
+    base="registry.suse.com/bci/bci-base:latest",
+    containerfile=GIT_SERVER_CONTAINERFILE,
+)
+
+GIT_POD = Pod(
+    containers=[
+        GIT_SERVER_CONTAINER,
+        container_from_pytest_param(GIT_CONTAINER),
+    ],
+)
+
+
+@pytest.mark.parametrize("pod_per_test", [GIT_POD], indirect=True)
+def test_git_clone_ssh(pod_per_test: PodData) -> None:
+    """Test that we can clone a repository over SSH successfully."""
+    srv = pod_per_test.container_data[0].connection
+    cli = pod_per_test.container_data[1].connection
+
+    cli.run_expect(
+        [0],
+        f'mkdir /root/.ssh && echo -e "{PRIV_KEY}" > /root/.ssh/id_ed25519 && chmod 0600 /root/.ssh/id_ed25519',
+    )
+    cli.run_expect([0], "eval `ssh-agent -s` && ssh-add /root/.ssh/id_ed25519")
+    cli.run_expect([0], "ssh-keyscan -H 127.0.0.1 >> /root/.ssh/known_hosts")
+
+    cli.run_expect(
+        [0], "git clone git@127.0.0.1:/srv/git/project.git /project"
+    )
+    cli.file("/project/.git/HEAD").exists
+    out = cli.check_output("git -C /project status")
```

Comment:
you can make these shorter by substituting `cli.run_expect([0], $cmd)` with `cli.check_output($cmd)`